components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/impl/ServiceDiscovererKubernetesTestCase.java

```diff
@@ -0,0 +1,355 @@
+                .withType("LoadBalancer").withClusterIP("1.1.1.2").withPorts(nodePort2).and()
+                .withNewStatus().withNewLoadBalancer().withIngress(ipIngresses).endLoadBalancer().and()
+                .build();
+
+        Service service3 = new ServiceBuilder().withNewMetadata()
+                .withName("service3").withNamespace("prod").withLabels(oneLabel).and()
+                .withNewSpec()
+                .withType("ClusterIP").withClusterIP("1.1.1.3").withPorts(httpPort).and()
+                .build();
+
+        Service service4 = new ServiceBuilder().withNewMetadata()
+                .withName("service4").withNamespace("prod").and()
+                .withNewSpec()
+                .withType("LoadBalancer").withClusterIP("1.1.1.4").withPorts(nodePort4).and()
+                .withNewStatus().withNewLoadBalancer().withIngress(ipIngresses).endLoadBalancer().and()
+                .build();
+
+        List<Service> servicesList = new ArrayList<>();
+        servicesList.add(service0);
+        servicesList.add(service1);
+        servicesList.add(service2);
+        servicesList.add(service3);
+        servicesList.add(service4);
+        this.listOfServices = servicesList;
+    }
+
+    private HashMap<String, String> createImplParametersMap(String externalTokenFileName) {
+        HashMap<String, String> implParameters = new HashMap<>();
+        implParameters.put(ServiceDiscovererKubernetes.EXTERNAL_SA_TOKEN_FILE_NAME, externalTokenFileName);
+        implParameters.put(ServiceDiscovererKubernetes.POD_MOUNTED_SA_TOKEN_FILE_PATH,
+                "/var/run/secrets/kubernetes.io/serviceaccount/token");
+        return implParameters;
+    }
+
+    private HashMap<String, String> createOneLabelHashMap() {
+        HashMap<String, String> oneLabel = new HashMap<>();
+        oneLabel.put("app", "web");
+        return oneLabel;
+    }
+
+    private ServiceList createServiceList() {
+        List<Service> servicesList = new ArrayList<>();
+        servicesList.add(listOfServices.get(0));
+        servicesList.add(listOfServices.get(1));
+        servicesList.add(listOfServices.get(2));
+        servicesList.add(listOfServices.get(3));
+        servicesList.add(listOfServices.get(4));
+        return new ServiceListBuilder().withItems(servicesList).build();
+    }
+
+    private ServiceList createServiceListWithNamespace() {
+        List<Service> servicesList = new ArrayList<>();
+        servicesList.add(listOfServices.get(0));
+        servicesList.add(listOfServices.get(1));
+        servicesList.add(listOfServices.get(2));
+        return new ServiceListBuilder().withItems(servicesList).build();
+    }
+
+    private ServiceList createServiceListWithCriteria() {
+        List<Service> servicesList = new ArrayList<>();
+        servicesList.add(listOfServices.get(0));
+        servicesList.add(listOfServices.get(3));
+        return new ServiceListBuilder().withItems(servicesList).build();
+    }
+
```

Comment:
Here some methods are missing method comments. Could you please add them?

Reply:
I have missed adding test descriptions. I'll add them all.